docs/containers-transports.5.md

```diff
@@ -60,7 +60,7 @@ The _algo:digest_ refers to the image ID reported by docker-inspect(1).
 ### **oci:**_path[:reference]_
 
 An image compliant with the "Open Container Image Layout Specification" at _path_.
-Using a _reference_ is optional and allows for storing multiple images at the same _path_.
+Using a _reference_ is optional for a single image reference but required for storing multiple images at the same _path_.
```

Comment:
This is not quite precise. It’s _accessing_ an image from a multi-image archive that requires providing a disambiguating name; the code will happily enough add (untagged/unnamed) images to an archive, as the motivating bug report demonstrates.

Reply:
@mtrmac would you prefer an additional sentence and/or an example in the docs for clarifying the github issue case? Something like `pulling multiple images without specification of reference will not preserve all references` ?

Reply:
> pulling multiple images without specification of reference will not preserve all references ?

I’m not quite sure what that means. (It’s a general principle of `skopeo copy` that names used in the source reference don’t affect names in the destination reference; it’s not something specific to this transport. So, copies by `skopeo copy` never “preserve” names/references in that sense — OTOH Podman _does_ have various heuristics of that kind.)

---

The general kinds information we probably want to convey for the two transports:
* (Maybe: The _reference_ value corresponds to the `org.opencontainers.image.ref.name` annotation.)
* When reading images from `oci:` or `oci-archive:`, if _path_ contains multiple images, a _reference_ must be specified to disambiguate; otherwise, the only image is implied.
* When writing images, _reference_ is optional and set on the written image [to allow disambiguating reads]
* Repeated image writes to `oci:` don’t overwrite pre-existing images, whether or not _reference_ is provided (but every _reference_ will only point to one image).
* Repeated image writes to `oci-archive:` overwrite the whole archive; only one image, the last written one, is included. (Do we want to _guarantee_ that? Probably, ambiguity doesn’t help anyone, but I’m not 100% sure.)

(Warning: I didn’t actually test any of this right now, just reading the code.)